File: rb/features.rb
# Transport SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module TransportFeatures
  def self.make_feature(name)
    case name
    when "base"
      TransportBaseFeature.new
    when "ratelimit"
      TransportRatelimitFeature.new
    when "retry"
      TransportRetryFeature.new
    when "test"
      TransportTestFeature.new
    when "timeout"
      TransportTimeoutFeature.new
    else
      TransportBaseFeature.new
    end
  end
end

File: py/features.py
# Transport SDK feature factory

from feature.base_feature import TransportBaseFeature
from feature.test_feature import TransportTestFeature


def _make_feature(name):
    features = {
        "base": lambda: TransportBaseFeature(),
        "test": lambda: TransportTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
